### PR TITLE
Custom Holoforms for pAIs and AIs oh and adds certain cyborg pai holograms. also mild pai nerf.

### PR DESCRIPTION
**Pull request by silicons**
## About The Pull Request

Adds custom holoforms, cloning your current preferences slot with your highest job (if any, otherwise you're nude yay congratulations you) and blending it with a filter of choice.
Use the two new client verbs to set it, 10 second cooldown because getFlatIcon is expensive as hell (but necessary for this to *properly* work unless I add overlays on the pAI/AI hologram holders themselves, which I'm too lazy to do)

oh and shifts carbon update_transform to living to reduce copypaste in paicode/weird edge cases.

oh and lets pais use sound emotes and stuff and yeah.

oh and nerfs pai cards to not be indestructible and a pai shorting out will need their radio manually reset, or automatically reset after 5 minutes.

oh and pais can buckle/unbuckle, but can no longer buckle others lmao why is that a thing.

## Why It's Good For The Game

customization and my snowflakes

## Changelog
:cl:
add: Custom holoforms have been added for pAIs and AIs. oh and cyborg holograms of specific modules too.
balance: pais are no longer indestructible-flagged while in card form. pai radios now short out if they are forcefully collapsed from damage.
/:cl:

### CODE REVIEW

**Comment by GrayRachnid** [comment]
how would it work if I have AI on high

**Comment by silicons** [comment]
@GrayRachnid it considers it as no job because ai/cyborg are blacklisted

**Comment by GrayRachnid** [comment]
So would it go after the medium preference then?

**Comment by silicons** [comment]
@GrayRachnid nope, that's a limitation it doesn't do the second highest/etc, it just goes for the highest if it's invalid it just assumes no job, I didn't want to complicate the get_highest_job() proc 

In the future I'm hoping someone with the time codes something to allow for custom outfits. I mean, I know enough about html to do it it's rather "what if a player chooses something that will cause issues if spawned, even in nullspace".

**Comment by ShadeAware** [comment]
**YOU MAD LAD, YOU ACTUALLY DID IT**

File: code/__HELPERS/custom_holoforms.dm [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+	var/icon/combined = new
+	combined.Insert(north, dir = NORTH)
+	CHECK_TICK
+	combined.Insert(south, dir = SOUTH)
+	CHECK_TICK
+	combined.Insert(east, dir = EAST)
+	CHECK_TICK
+	combined.Insert(west, dir = WEST)
+	CHECK_TICK
```
Comment (Ghommie) [inline code/__HELPERS/custom_holoforms.dm]: Wouldn't it be better if it was done like this?:
```
		var/icon/combined= new
		for(var/D in GLOB.cardinals)
			mannequin.setDir(D)
			COMPILE_OVERLAYS(mannequin)
			CHECK_TICK
			var/icon/partial = getFlatIcon(mannequin)
			CHECK_TICK
			combined.Insert(partial, dir=D)
			CHECK_TICK
```

[Reference](https://github.com/Citadel-Station-13/Citadel-Station-13/blob/7fa485201a4e62d12f8a3930611477b1d21c4f38/code/__HELPERS/icons.dm#L1059)

File: modular_citadel/code/modules/mob/cit_emotes.dm [hunk 2]
```diff
@@ -87,7 +86,7 @@
 	emote_type = EMOTE_AUDIBLE
 
 /datum/emote/living/snap/run_emote(mob/living/user, params)
-	if(ishuman(user))
+	if(iscarbon(user) || ispAI(user))
```
Comment (Ghommie) [inline modular_citadel/code/modules/mob/cit_emotes.dm]: Shitty citadel emotes.
Move the parent call on top, then add an early return if **.** is FALSE.
Then remove this typecheck.

File: modular_citadel/code/modules/mob/cit_emotes.dm [hunk 3]
```diff
@@ -101,7 +100,7 @@
 	emote_type = EMOTE_AUDIBLE
 
 /datum/emote/living/snap2/run_emote(mob/living/user, params)
-	if(ishuman(user))
+	if(iscarbon(user) || ispAI(user))
```
Comment (Ghommie) [inline modular_citadel/code/modules/mob/cit_emotes.dm]: Same.

File: modular_citadel/code/modules/mob/cit_emotes.dm [hunk 4]
```diff
@@ -115,7 +114,7 @@
 	emote_type = EMOTE_AUDIBLE
 
 /datum/emote/living/snap3/run_emote(mob/living/user, params)
-	if(ishuman(user))
+	if(iscarbon(user) || ispAI(user))
```
Comment (Ghommie) [inline modular_citadel/code/modules/mob/cit_emotes.dm]: Same.

File: modular_citadel/code/modules/mob/cit_emotes.dm [hunk 5]
```diff
@@ -129,7 +128,7 @@
 	emote_type = EMOTE_AUDIBLE
 
 /datum/emote/living/awoo/run_emote(mob/living/user, params)
-	if(ishuman(user))
+	if(iscarbon(user) || ispAI(user))
```
Comment (Ghommie) [inline modular_citadel/code/modules/mob/cit_emotes.dm]: Idem.

File: modular_citadel/code/modules/mob/cit_emotes.dm [hunk 6]
```diff
@@ -143,7 +142,7 @@
 	emote_type = EMOTE_AUDIBLE
 
 /datum/emote/living/nya/run_emote(mob/living/user, params)
-	if(ishuman(user))
+	if(iscarbon(user) || ispAI(user))
```
Comment (Ghommie) [inline modular_citadel/code/modules/mob/cit_emotes.dm]: As usual.

File: modular_citadel/code/modules/mob/cit_emotes.dm [hunk 7]
```diff
@@ -157,7 +156,7 @@
 	emote_type = EMOTE_AUDIBLE
 
 /datum/emote/living/weh/run_emote(mob/living/user, params)
-	if(ishuman(user))
+	if(iscarbon(user) || ispAI(user))
```
Comment (Ghommie) [inline modular_citadel/code/modules/mob/cit_emotes.dm]: Look all this copypasta.

File: modular_citadel/code/modules/mob/cit_emotes.dm [hunk 8]
```diff
@@ -171,7 +170,7 @@
 	emote_type = EMOTE_AUDIBLE
 
 /datum/emote/living/peep/run_emote(mob/living/user, params)
-	if(ishuman(user))
+	if(iscarbon(user) || ispAI(user))
```
Comment (Ghommie) [inline modular_citadel/code/modules/mob/cit_emotes.dm]: This says much about the low quality of citadel.

File: modular_citadel/code/modules/mob/cit_emotes.dm [hunk 9]
```diff
@@ -190,10 +189,9 @@
 	key_third_person = "lets out a tiny squeak"
 	message = "lets out a tiny squeak!"
 	emote_type = EMOTE_AUDIBLE
-	mob_type_allowed_typecache = list(/mob/living/carbon)
 
 /datum/emote/living/mothsqueak/run_emote(mob/living/user, params)
-	if(ishuman(user))
+	if(iscarbon(user) || ispAI(user))
```
Comment (Ghommie) [inline modular_citadel/code/modules/mob/cit_emotes.dm]: I regret removing the regression label for a moment.

File: modular_citadel/code/modules/mob/cit_emotes.dm [hunk 10]
```diff
@@ -207,7 +205,7 @@
 	emote_type = EMOTE_AUDIBLE
 
 /datum/emote/living/merp/run_emote(mob/living/user, params)
-	if(ishuman(user))
+	if(iscarbon(user) || ispAI(user))
```
Comment (Ghommie) [inline modular_citadel/code/modules/mob/cit_emotes.dm]: Despite the fact `mob_type_allowed_typecache` existed for a reason, you had to do this instead.

File: modular_citadel/code/modules/mob/cit_emotes.dm [hunk 11]
```diff
@@ -221,7 +219,7 @@
 	emote_type = EMOTE_AUDIBLE
 
 /datum/emote/living/bark/run_emote(mob/living/user, params)
-	if(ishuman(user))
+	if(iscarbon(user) || ispAI(user))
```
Comment (Ghommie) [inline modular_citadel/code/modules/mob/cit_emotes.dm]: It makes no sense for non-carbons / non-pAI to be able to bark silently either. But I'm not here to tell you to add dogs to this one's `mob_type_allowed_typecache`. Or am I?

File: modular_citadel/code/modules/mob/cit_emotes.dm [hunk 12]
```diff
@@ -77,8 +77,7 @@
 	restraint_check = TRUE
 	emote_type = EMOTE_AUDIBLE
 	sound = 'sound/effects/snap01.ogg'
-	mob_type_allowed_typecache = list(/mob/living/carbon/)
-
+	mob_type_allowed_typecache = list(/mob/living/carbon, /mob/living/silicon/pai)
```
Comment (Ghommie) [inline modular_citadel/code/modules/mob/cit_emotes.dm]: Now, once you have replaced all those typechecks with proper parent return value checks, copy this line to each following typepath too, thank you.

**Review by Ghommie — CHANGES_REQUESTED** [review]
Resolve the followings correctly and It's good to merge.

**Review by Ghommie — APPROVED** [review]

**Pull request merged**